File: python_converted/src/api/controllers/endpoints.py
from typing import Any, Dict


const API_ENDPOINTS = {
  AUTH: Dict[str, Any],
  USER: Dict[str, Any],
  MAP: Dict[str, Any]`,
    UPDATE: (id: str) => `/maps/${id}`,
    DELETE: (id: str) => `/maps/${id}`,
    REGIONS: (mapId: str) => `/maps/${mapId}/regions`,
  },
  REGION: Dict[str, Any]/regions`,
    CREATE: (mapId: str) => `/maps/${mapId}/regions`,
    GET: (mapId: str, regionId: str) => `/maps/${mapId}/regions/${regionId}`,
    UPDATE: (mapId: str, regionId: str) => `/maps/${mapId}/regions/${regionId}`,
    DELETE: (mapId: str, regionId: str) => `/maps/${mapId}/regions/${regionId}`,
  },
} as const

File: src/api/controllers/endpoints.ts
// API Endpoints
export const API_ENDPOINTS = {
  // Auth endpoints
  AUTH: {
    LOGIN: '/auth/login',
    REGISTER: '/auth/register',
    LOGOUT: '/auth/logout',
    REFRESH_TOKEN: '/auth/refresh-token',
  },

  // User endpoints
  USER: {
    PROFILE: '/user/profile',
    UPDATE_PROFILE: '/user/profile/update',
    CHANGE_PASSWORD: '/user/password/change',
  },

  // Map endpoints
  MAP: {
    LIST: '/maps',
    CREATE: '/maps',
    GET: (id: string) => `/maps/${id}`,
    UPDATE: (id: string) => `/maps/${id}`,
    DELETE: (id: string) => `/maps/${id}`,
    REGIONS: (mapId: string) => `/maps/${mapId}/regions`,
  },

  // Region endpoints
  REGION: {
    LIST: (mapId: string) => `/maps/${mapId}/regions`,
    CREATE: (mapId: string) => `/maps/${mapId}/regions`,
    GET: (mapId: string, regionId: string) => `/maps/${mapId}/regions/${regionId}`,
    UPDATE: (mapId: string, regionId: string) => `/maps/${mapId}/regions/${regionId}`,
    DELETE: (mapId: string, regionId: string) => `/maps/${mapId}/regions/${regionId}`,
  },
} as const;
